addons/mail/static/src/core/common/suggestion_service.js

```diff
@@ -294,13 +318,17 @@ export class SuggestionService {
         return {};
     }
 
+    filterChannelSuggestions(channels) {
+        return channels.filter((channel) => channel.channel_type === "channel");
+    }
+
```

Comment:
Don't like this filter neither: adds an unnecessary level of indirection.
I would just filter immediately where needed.

More functions are more unnecessary complexity. And don't forget we want to replace services with JS models to make code DRYer.